controllers/internal/volsnapshot/vol_snapshot_control.go

```diff
@@ -0,0 +1,119 @@
+package volsnapshot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	cosmosalpha "github.com/strangelove-ventures/cosmos-operator/api/v1alpha1"
+	"github.com/strangelove-ventures/cosmos-operator/controllers/internal/fullnode"
+	"github.com/strangelove-ventures/cosmos-operator/controllers/internal/kube"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Client is a subset of client.Client.
+type Client interface {
+	Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error
+	List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+}
+
+type PodFinder interface {
+	SyncedPod(ctx context.Context, candidates []*corev1.Pod) (*corev1.Pod, error)
+}
+
+// VolumeSnapshotControl manages VolumeSnapshots
+type VolumeSnapshotControl struct {
+	client Client
+	finder PodFinder
+	now    func() time.Time
+}
+
+func NewVolumeSnapshotControl(client Client, finder PodFinder) *VolumeSnapshotControl {
+	return &VolumeSnapshotControl{
+		client: client,
+		finder: finder,
+		now:    time.Now,
+	}
+}
+
+// Candidate is a target instance of a CosmosFullNode from which to make a snapshot.
+type Candidate struct {
+	PodLabels map[string]string
+	PodName   string
+	PVCName   string
+}
+
+// FindCandidate finds a suitable candidate for creating a volume snapshot.
+// Any errors returned can be treated as transient; worth a retry.
+func (control VolumeSnapshotControl) FindCandidate(ctx context.Context, crd *cosmosalpha.ScheduledVolumeSnapshot) (Candidate, error) {
```

Comment:
This method + tests were already completed in a previous PR. The file rename is making them appear as if it's new code.